service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,24 +365,70 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
+  @WriteTransaction
+  private boolean updateResource(
+      UUID workspaceId, UUID resourceId, Map<String, String> updateParams) {
+
+    var params = new MapSqlParameterSource();
+
+    updateParams.forEach(
+        (k, v) -> {
+          if (!StringUtils.isEmpty(k) && !StringUtils.isEmpty(v)) {
+            params.addValue(k, v);
+          }
+        });
+
+    return updateResourceColumns(workspaceId, resourceId, params);
+  }
+
+  /**
+   * Update name, description, and/or attributes of the resource.
+   *
+   * @param name name of the resource, may be null if it does not need to be updated
+   * @param description description of the resource, may be null if it does not need to be updated
+   * @param attributes resource-type specific attributes, may be null if it does not need to be
+   *     updated .
+   */
   @WriteTransaction
   public boolean updateResource(
-      UUID workspaceId, UUID resourceId, String name, String description) {
-    if (name == null && description == null) {
+      UUID workspaceId,
+      UUID resourceId,
+      @Nullable String name,
+      @Nullable String description,
+      @Nullable String attributes) {
+    if (name == null & description == null && attributes == null) {
       return false;
     }
+    Map<String, String> updateParams = getUpdateParams(name, description, attributes);
 
-    var params = new MapSqlParameterSource();
+    return updateResource(workspaceId, resourceId, updateParams);
+  }
 
-    if (name != null) {
-      params.addValue("name", name);
+  /**
+   * Update name and/or description of the resource.
+   *
+   * @param name name of the resource, may be null if it does not need to be updated
+   * @param description description of the resource, may be null if it does not need to be updated
+   */
+  @WriteTransaction
+  public boolean updateResource(
+      UUID workspaceId, UUID resourceId, @Nullable String name, @Nullable String description) {
+    if (name == null & description == null) {
+      return false;
     }
+    Map<String, String> updateParams = getUpdateParams(name, description, null);
 
-    if (description != null) {
-      params.addValue("description", description);
-    }
+    return updateResource(workspaceId, resourceId, updateParams);
+  }
 
-    return updateResourceColumns(workspaceId, resourceId, params);
+  /** Gets a map of the params that needs to be updated. */
+  private static Map<String, String> getUpdateParams(
+      @Nullable String name, @Nullable String description, @Nullable String attributes) {
+    Map<String, String> updateParams = new HashMap<>();
```

Comment:
minor: given that there are only three things, it seems like a lot of effort to build a map that tells you how to build another map.